### DISCUSSION TITLE
Handle the function of changing the color of the layer when clicking on that layer on the map

### DISCUSSION THREAD

**Discussion by hoangdangcodecomplete** [tomik23/react-leaflet-examples#12] [Q&A]
Dear Grzegorz Tomicki,

Currently, I'm working on a map product and the library I'm using is React Leaflet.

In my map I have polygon and polyline drawing action and I used react-leaflet-draw library for that. Currently, what I want most in my map is when I finish drawing those shapes, I can change the colors of those drawings and change the dashArray property of each shape when the user selects any shape on the map.

Here is my project: [Link code](https://github.com/hoangdangcodecomplete/map-clone)
Here is the demo of my map: [Video](https://youtu.be/3-ENGmGeWwM)

Please help me write a small function in my code so I can change the color of the drawings when clicking on it and store it with a list of data so that I can send it back using the API. 
I really thank you for that.

If you have any questions about the code or the map let me know and I will answer them for you.

Once again, thank you so much, and hope you have a nice day!

**Comment by tomickigrzegorz** [comment]
Unfortunately, I can't help. No time and what you want to do is a big topic.
I do so by first creating a given element with default colors. Later when I click on polygon 
![edit](https://user-images.githubusercontent.com/25438490/175244328-5e5ff3a5-63b9-402f-b119-6be0ffd7d4e6.png)
```
polgonName.on("click", funciton(e) {
  popupCreate(e.layer);
});
```
I create a popup with the form to which I pass e.layer
Each color change or opacity changes 
```
layer.feature.properties = {
   ...layer.feature.properties,
   color: color.value
}
```
Of course, I do it because all changes are saved in localStorage, and that's exactly what geojson saves and changes to localStorage in order to be able to work on objects later when I close the browser.
In the current project, I have the ability to export gojson, load and show on the map and many options.

Hope I helped a little.

Good luck

**Comment by hoangdangcodecomplete** [comment]
@tomik23  Really thank you for this help.
I have found the solution as well but I am having a problem here can you help me a bit ?
This is my issue: https://stackoverflow.com/questions/72723990/how-to-change-the-color-of-the-layer-when-click-in-leaflet-react
Sorry for bothering your time but I am also trying to solve it hope you will help me.
